delivery_tnt_oca/models/tnt_request.py

```diff
@@ -192,32 +206,36 @@ def _prepare_sender(self):
         return data
 
     def _get_data_total_shipping(self):
-        weight = self.record.shipping_weight
-        volume = p_length = height = width = 0
-        if self.record.package_ids:
-            weight = sum(self.record.package_ids.mapped("quant_ids.quantity"))
-            height = max(self.record.package_ids.mapped("height"))
-            width = max(self.record.package_ids.mapped("width"))
-            p_length = max(self.record.package_ids.mapped("length"))
-            for quant in self.record.package_ids.mapped("quant_ids"):
-                volume += quant.product_id.volume * quant.quantity
+        if self.use_packages_from_picking and self.record.package_ids:
+            weight = sum(
+                [p.shipping_weight or p.weight for p in self.record.package_ids]
+            )
+            height = sum(self.record.package_ids.mapped("height"))
+            width = sum(self.record.package_ids.mapped("width"))
+            p_length = sum(self.record.package_ids.mapped("length"))
         else:
-            lines = self.record.move_line_ids_without_package
-            for line in lines.filtered(lambda x: x.qty_done > 0):
-                volume += line.product_id.volume * line.qty_done
-                p_length += line.product_id.product_length * line.qty_done
-                height += line.product_id.product_height * line.qty_done
-                width += line.product_id.product_width * line.qty_done
+            weight = self.record.shipping_weight
+            # in cm, we need to convert to m
```

Comment:
Where is stated that this is in cm?

Reply:
Actually this is used ONLY if packages are not used in `stock.picking` (`package_ids`) or you do not want to use the picking packages (`use_packages_from_picking` field of the carrier).
Really only cm is specified in the name of the `product.packing` (_TNT Default (cm)_) that will be used as the default package.

What other option do you propose?